packages/vm/src/context.rs

```diff
@@ -362,112 +379,106 @@ mod test {
         );
 
         // now is empty again
-        let (ends, endq) = move_out_of_context::<MS, MQ>(ctx);
+        let (ends, endq) = move_out_of_context::<MS, MQ>(&env);
         assert!(ends.is_none());
         assert!(endq.is_none());
     }
 
     #[test]
-    #[cfg(feature = "default-singlepass")]
     fn gas_tracking_works_correctly() {
-        let mut instance = make_instance();
+        let (env, _instance) = make_instance();
 
         let gas_limit = 100;
-        set_gas_left(instance.context_mut(), gas_limit);
-        get_gas_state_mut::<MS, MQ>(instance.context_mut()).set_gas_limit(gas_limit);
-        let context = instance.context_mut();
+        set_gas_left(&env, gas_limit);
+        env.with_gas_state_mut(|state| state.set_gas_limit(gas_limit));
 
         // Consume all the Gas that we allocated
-        account_for_externally_used_gas::<MS, MQ>(context, 70).unwrap();
-        account_for_externally_used_gas::<MS, MQ>(context, 4).unwrap();
-        account_for_externally_used_gas::<MS, MQ>(context, 6).unwrap();
-        account_for_externally_used_gas::<MS, MQ>(context, 20).unwrap();
+        account_for_externally_used_gas::<MS, MQ>(&env, 70).unwrap();
```

Comment:
You use the same `&env` in multiple calls here. Odd this doesn't work above?

Reply:
Oh, nice find, it does work: 2f8f86dfd7c3e4ad0d31a82d9dfa43b06c0ca5e6

It did not work before, then `Env` was mutable and we used mutable references. But this changed to always immutable with internal mutability.